commit 21c9ffcad820e9b4c06dbcf263a35b67465d7ee6 (refs/changes/12/5112/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-11-22T17:53:16-08:00 (11 months ago)
    
    MANTA-4030 many images using node 0.10 or 0.12 broken by transitive lru-cache@4.1.4 dep
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 3cf065c..e34bf8a 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -8,6 +8,12 @@
 
 ## Not yet released
 
+## 12.1.1
+
+- MANTA-4030 Drop unused lru-cache dep (and internal "lib/cache.js" lib) to
+  allow sdc-clients users to work around the transitive lru-cache@4.1.4
+  breakage to node 0.10 usage.
+
 ## 12.1.0
 
 - TRITON-774 Add imgapi.addImageFileFromUrl method, used to add a image file
diff --git a/lib/cache.js b/lib/cache.js
deleted file mode 100644
index 35136fd..0000000
--- a/lib/cache.js
+++ /dev/null
@@ -1,73 +0,0 @@
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
-/*
- * Copyright (c) 2014, Joyent, Inc.
- */
-
-var assert = require('assert');
-
-var LRU = require('lru-cache');
-
-
-
-// --- API
-
-function Cache(options) {
-    if (typeof (options) !== 'object')
-        throw new TypeError('options (object) required');
-
-    this._cache = LRU(options.size || 1000);
-    this._age = (options.age || 60) * 1000;
-
-    var self = this;
-    this.__defineGetter__('age', function () { return self._age / 1000; });
-    this.__defineSetter__('age', function (a) {
-        self._age = a * 1000;
-    });
-}
-
-
-Cache.prototype.get = function (key) {
-    assert.ok(key);
-
-    var entry = this._cache.get(key);
-    if (!entry)
-        return null;
-
-    var now = new Date();
-    if ((now.getTime() - entry.ctime) > this._age)
-        return null;
-
-    return entry.value;
-};
-
-
-Cache.prototype.put = function (key, value) {
-    assert.ok(key);
-
-    var entry = {
-        ctime: new Date().getTime(),
-        value: value
-    };
-
-    this._cache.set(key, entry);
-    return value;
-};
-
-
-
-// --- Exports
-
-module.exports = {
-
-    createCache: function (options) {
-        return new Cache(options || {});
-    },
-
-    Cache: Cache
-
-};
diff --git a/package.json b/package.json
index 33e01d4..7209e02 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdc-clients",
   "description": "node.js client libraries for Triton core REST APIs.",
-  "version": "12.1.0",
+  "version": "12.1.1",
   "homepage": "http://www.joyent.com",
   "repository": {
     "type": "git",
@@ -15,7 +15,6 @@
     "bunyan": "^1.8.1",
     "clone": "0.1.8",
     "jsprim": "1.4.1",
-    "lru-cache": "2.3.0",
     "lomstream": "1.1.0",
     "once": "^1.3.1",
     "restify-clients": "^1.6.0",
